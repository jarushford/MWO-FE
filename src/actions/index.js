export const setUser = user => ({
  type: 'SET_USER',
  user
})

export const setTourDates = dates => ({
  type: 'SET_TOUR_DATES',
  dates
})

export const setVideos = videos => ({
  type: 'SET_VIDEOS',
  videos
})

export const setPhotos = photos => ({
  type: 'SET_PHOTOS',
  photos
})

export const setNews = news => ({
  type: 'SET_NEWS',
  news
})