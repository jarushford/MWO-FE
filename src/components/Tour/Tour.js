import React, { Component } from 'react'
import '../../main.scss'

export default class Tour extends Component {
  constructor() {
    super()
    this.state = {
      tourDates: [],
      isLoading: false
    }
  }

  componentDidMount() {
   this.getTourDates() 
  }

  getTourDates = async () => {
    try {
      this.setState({ isLoading: true })
      const url = 'https://mwo-be.herokuapp.com/api/v1/tour_dates'
      const response = await fetch(url)

      if (!response.ok) {
        // set error action
      } else {
        const tourDates = await response.json()
        this.setState({ tourDates, isLoading: false })
      }
    } catch (error) {
      // set error action
    }
  }

  render() {
    let noDates
    let tourDates

    if (this.state.isLoading) {
      noDates = <h3>Loading ...</h3>
      // use react loader later
    } else if (!this.state.tourDates.length) {
      noDates = <h3>No tour dates right now, check back soon!</h3>
    } else {
      tourDates = this.state.tourDates.map(date => {
        return (
          <tr key={date.id}>
            <td>{date.day_of_week} {date.date}</td>
            <td>{date.venue}</td>
            <td>{date.city}</td>
            <td>
              <a href={date.ticket_link}>
                <button>Tickets</button>
              </a>
            </td>
          </tr>
        )
      })
    }

    const tableHeaders = ['when', 'where', 'wait... where?', 'how']

    return (
      <section className="tour-container">
        <table>
          <thead>
            <tr>
              {
                tableHeaders.map(header => {
                  return <th key={header}>{header}</th>
                })
              }
            </tr>
          </thead>
          <tbody>
            {tourDates}
          </tbody>
        </table>
        {noDates}
      </section>
    )
  }
}