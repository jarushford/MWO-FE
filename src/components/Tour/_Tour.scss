.tour-container {
  width: 100%;
  margin: 0;

  h1 {
    color: $lightOverlay;
    font-size: 15rem;
    margin: 0;
    opacity: .35;
    height: 175px;
    line-height: 150px;
  }

  table {
    width: 95%;
    margin: 20px auto 100px;
  }

  h3, td, th {
    color: white;
    font-family: 'Open Sans', sans-serif;
    font-weight: 400;
    text-align: center;
  }

  h3 {
    margin-top: -20px;
    margin-bottom: 100px;
    padding: 0 30px;
  }

  td {
    height: 60px;
    border-bottom: 1px solid $lightOverlay;
  }

  th {
    font-size: 1.5rem;
    border-bottom: 2px solid $lightOverlay;
  }
}

.tickets-btn {
  width: 80px;
  height: 40px;
  background: none;
  border: 2px solid $purple;
  border-radius: 5px;
  color: $purple;
  font-family: 'Open Sans', sans-serif;
  font-size: 1rem;
  transition: .3s ease;
  outline: none;

  &:hover {
    background: $purple;
    color: white;
  }
}

@media screen and (max-width: 600px) {
  .tour-container {
    th {
      font-size: 1rem;
    }

    td {
      font-size: .7rem;
    }
  }

  .tickets-btn {
    font-size: .7rem;
    width: 60px;
    height: 30px;
  }
}