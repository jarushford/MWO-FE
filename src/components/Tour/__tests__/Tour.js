import React from 'react'
import { shallow } from 'enzyme'
import { mapStateToProps, mapDispatchToProps, Tour } from '../Tour'
import { setTourDates } from '../../../actions'

describe('Tour', () => {
  describe('mapStateToProps', () => {
    it('should return a props object with data from the store', () => {
      const state = {
        tourDates: [],
        otherThings: 'sdfkjhlsief'
      }
      const expected = {
        tourDates: []
      }

      const result = mapStateToProps(state)

      expect(result).toEqual(expected)
    })
  })

  describe('mapDispatchToProps', () => {
    it('should return a props object with a method setTourDates', () => {
      const mockDispatch = jest.fn()
      const expected = setTourDates([])

      const mappedProps = mapDispatchToProps(mockDispatch)
      mappedProps.setTourDates([])

      expect(mockDispatch).toBeCalledWith(expected)
    })
  })

  describe('Tour component', () => {
    let wrapper, tourDates, setTourDates

    beforeEach(() => {
      setTourDates = jest.fn()
      tourDates = [
        {
          id: 123,
          day_of_week: 'Friday',
          date: '12/12/1212',
          venue: 'Place',
          city: 'Town',
          ticket_link: 'isfs4fijrhxdff'
        },
        {
          id: 123,
          day_of_week: 'Friday',
          date: '12/23/1212',
          venue: 'Place',
          city: 'Town',
          ticket_link: 'isfs4fijrhxdff'
        }
      ]
      wrapper = shallow(<Tour 
        tourDates={tourDates}
        setTourDates={setTourDates}
      />, { disableLifecycleMethods: true })
    })

    it('should match the default snapshot', () => {
      expect(wrapper).toMatchSnapshot()
    })

    it('should match the snapshot if errored', () => {
      wrapper.setState({ error: 'Bad stuff!' })

      expect(wrapper).toMatchSnapshot()
    })

    it('should match the snapshot if loading', () => {
      wrapper.setState({ isLoading: true })

      expect(wrapper).toMatchSnapshot()
    })

    it('should match the snapshot with no tour dates', () => {
      wrapper = shallow(<Tour 
        tourDates={[]}
        setTourDates={setTourDates}
      />, { disableLifecycleMethods: true })

      expect(wrapper).toMatchSnapshot()
    })

    it('should have the correct default state', () => {
      const expected = {
        tourDates: [],
        isLoading: false,
        error: ''
      }

      expect(wrapper.state()).toEqual(expected)
    })

    describe('componentDidMount', () => {
      it('should setState of tourdates if they are in the store already', () => {
        wrapper.instance().componentDidMount()

        expect(wrapper.state().tourDates).toEqual(tourDates)
      })

      it('should call getTourDates if they are not in the store already', () => {
        wrapper = shallow(<Tour 
          tourDates={[]}
          setTourDates={setTourDates}
        />, { disableLifecycleMethods: true })

        const spy = jest.spyOn(wrapper.instance(), 'getTourDates')

        wrapper.instance().componentDidMount()

        expect(spy).toBeCalled()
      })
    })

    describe('getTourDates', () => {
      it('should setState of error if response is not ok', async () => {
        window.fetch = jest.fn().mockImplementation(() => Promise.resolve({
          ok: false,
          status: 400,
          statusText: 'Failure'
        }))

        await wrapper.instance().getTourDates()

        expect(wrapper.state().error).toEqual('400 Failure')
      })

      it('should setState of error if an unhandled server error occurs', async () => {
        window.fetch = jest.fn().mockImplementation(() => {
          throw Error('Ahhh!')
        })

        await wrapper.instance().getTourDates()

        expect(wrapper.state().error).toEqual(Error('Ahhh!'))
      })

      it('should setState of photos if response is ok', async () => {
        window.fetch = jest.fn().mockImplementation(() => Promise.resolve({
          ok: true,
          json: () => []
        }))

        await wrapper.instance().getTourDates()

        expect(wrapper.state().tourDates).toEqual([])
        expect(setTourDates).toBeCalledWith([])
      })
    })

    describe('getDatesRender', () => {
      it('should return the correct JSX for the tour dates in state', () => {
        wrapper.instance().componentDidMount()

        const dates = wrapper.instance().getDatesRender()

        expect(typeof dates).toEqual('object')
      })
    })

    describe('dateHelper', () => {
      it('should return a correctly formatted date string for sorting', () => {
        const date = '12/23'

        const result = wrapper.instance().dateHelper(date)

        expect(result).toEqual('1223')
      })
    })
  })
})