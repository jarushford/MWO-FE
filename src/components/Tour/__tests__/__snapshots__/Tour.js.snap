// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tour Tour component should match the default snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Tour
    setTourDates={[MockFunction]}
    tourDates={
      Array [
        Object {
          "city": "Town",
          "date": "12/12/1212",
          "day_of_week": "Friday",
          "id": 123,
          "ticket_link": "isfs4fijrhxdff",
          "venue": "Place",
        },
        Object {
          "city": "Town",
          "date": "12/23/1212",
          "day_of_week": "Friday",
          "id": 123,
          "ticket_link": "isfs4fijrhxdff",
          "venue": "Place",
        },
      ]
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <h1>
          tour
        </h1>,
        <table
          cellPadding="10"
          cellSpacing="0"
        >
          <thead>
            <tr>
              <th>
                when
              </th>
              <th>
                where
              </th>
              <th>
                wait... where?
              </th>
              <th>
                how
              </th>
            </tr>
          </thead>
          <tbody />
        </table>,
        <h3>
          No tour dates right now, check back soon!
        </h3>,
      ],
      "className": "tour-container",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "tour",
        },
        "ref": null,
        "rendered": "tour",
        "type": "h1",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "cellPadding": "10",
          "cellSpacing": "0",
          "children": Array [
            <thead>
              <tr>
                <th>
                  when
                </th>
                <th>
                  where
                </th>
                <th>
                  wait... where?
                </th>
                <th>
                  how
                </th>
              </tr>
            </thead>,
            <tbody />,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <tr>
                <th>
                  when
                </th>
                <th>
                  where
                </th>
                <th>
                  wait... where?
                </th>
                <th>
                  how
                </th>
              </tr>,
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <th>
                    when
                  </th>,
                  <th>
                    where
                  </th>,
                  <th>
                    wait... where?
                  </th>,
                  <th>
                    how
                  </th>,
                ],
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": "when",
                  "nodeType": "host",
                  "props": Object {
                    "children": "when",
                  },
                  "ref": null,
                  "rendered": "when",
                  "type": "th",
                },
                Object {
                  "instance": null,
                  "key": "where",
                  "nodeType": "host",
                  "props": Object {
                    "children": "where",
                  },
                  "ref": null,
                  "rendered": "where",
                  "type": "th",
                },
                Object {
                  "instance": null,
                  "key": "wait... where?",
                  "nodeType": "host",
                  "props": Object {
                    "children": "wait... where?",
                  },
                  "ref": null,
                  "rendered": "wait... where?",
                  "type": "th",
                },
                Object {
                  "instance": null,
                  "key": "how",
                  "nodeType": "host",
                  "props": Object {
                    "children": "how",
                  },
                  "ref": null,
                  "rendered": "how",
                  "type": "th",
                },
              ],
              "type": "tr",
            },
            "type": "thead",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": undefined,
            },
            "ref": null,
            "rendered": null,
            "type": "tbody",
          },
        ],
        "type": "table",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "No tour dates right now, check back soon!",
        },
        "ref": null,
        "rendered": "No tour dates right now, check back soon!",
        "type": "h3",
      },
    ],
    "type": "section",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <h1>
            tour
          </h1>,
          <table
            cellPadding="10"
            cellSpacing="0"
          >
            <thead>
              <tr>
                <th>
                  when
                </th>
                <th>
                  where
                </th>
                <th>
                  wait... where?
                </th>
                <th>
                  how
                </th>
              </tr>
            </thead>
            <tbody />
          </table>,
          <h3>
            No tour dates right now, check back soon!
          </h3>,
        ],
        "className": "tour-container",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "tour",
          },
          "ref": null,
          "rendered": "tour",
          "type": "h1",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "cellPadding": "10",
            "cellSpacing": "0",
            "children": Array [
              <thead>
                <tr>
                  <th>
                    when
                  </th>
                  <th>
                    where
                  </th>
                  <th>
                    wait... where?
                  </th>
                  <th>
                    how
                  </th>
                </tr>
              </thead>,
              <tbody />,
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <tr>
                  <th>
                    when
                  </th>
                  <th>
                    where
                  </th>
                  <th>
                    wait... where?
                  </th>
                  <th>
                    how
                  </th>
                </tr>,
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": Array [
                    <th>
                      when
                    </th>,
                    <th>
                      where
                    </th>,
                    <th>
                      wait... where?
                    </th>,
                    <th>
                      how
                    </th>,
                  ],
                },
                "ref": null,
                "rendered": Array [
                  Object {
                    "instance": null,
                    "key": "when",
                    "nodeType": "host",
                    "props": Object {
                      "children": "when",
                    },
                    "ref": null,
                    "rendered": "when",
                    "type": "th",
                  },
                  Object {
                    "instance": null,
                    "key": "where",
                    "nodeType": "host",
                    "props": Object {
                      "children": "where",
                    },
                    "ref": null,
                    "rendered": "where",
                    "type": "th",
                  },
                  Object {
                    "instance": null,
                    "key": "wait... where?",
                    "nodeType": "host",
                    "props": Object {
                      "children": "wait... where?",
                    },
                    "ref": null,
                    "rendered": "wait... where?",
                    "type": "th",
                  },
                  Object {
                    "instance": null,
                    "key": "how",
                    "nodeType": "host",
                    "props": Object {
                      "children": "how",
                    },
                    "ref": null,
                    "rendered": "how",
                    "type": "th",
                  },
                ],
                "type": "tr",
              },
              "type": "thead",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": undefined,
              },
              "ref": null,
              "rendered": null,
              "type": "tbody",
            },
          ],
          "type": "table",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "No tour dates right now, check back soon!",
          },
          "ref": null,
          "rendered": "No tour dates right now, check back soon!",
          "type": "h3",
        },
      ],
      "type": "section",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    "disableLifecycleMethods": true,
  },
}
`;

exports[`Tour Tour component should match the snapshot if errored 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Tour
    setTourDates={[MockFunction]}
    tourDates={
      Array [
        Object {
          "city": "Town",
          "date": "12/12/1212",
          "day_of_week": "Friday",
          "id": 123,
          "ticket_link": "isfs4fijrhxdff",
          "venue": "Place",
        },
        Object {
          "city": "Town",
          "date": "12/23/1212",
          "day_of_week": "Friday",
          "id": 123,
          "ticket_link": "isfs4fijrhxdff",
          "venue": "Place",
        },
      ]
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "message": "Bad stuff!",
    },
    "ref": null,
    "rendered": null,
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "message": "Bad stuff!",
      },
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    "disableLifecycleMethods": true,
  },
}
`;

exports[`Tour Tour component should match the snapshot if loading 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Tour
    setTourDates={[MockFunction]}
    tourDates={
      Array [
        Object {
          "city": "Town",
          "date": "12/12/1212",
          "day_of_week": "Friday",
          "id": 123,
          "ticket_link": "isfs4fijrhxdff",
          "venue": "Place",
        },
        Object {
          "city": "Town",
          "date": "12/23/1212",
          "day_of_week": "Friday",
          "id": 123,
          "ticket_link": "isfs4fijrhxdff",
          "venue": "Place",
        },
      ]
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {},
    "ref": null,
    "rendered": null,
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {},
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    "disableLifecycleMethods": true,
  },
}
`;

exports[`Tour Tour component should match the snapshot with no tour dates 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Tour
    setTourDates={[MockFunction]}
    tourDates={Array []}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <h1>
          tour
        </h1>,
        <table
          cellPadding="10"
          cellSpacing="0"
        >
          <thead>
            <tr>
              <th>
                when
              </th>
              <th>
                where
              </th>
              <th>
                wait... where?
              </th>
              <th>
                how
              </th>
            </tr>
          </thead>
          <tbody />
        </table>,
        <h3>
          No tour dates right now, check back soon!
        </h3>,
      ],
      "className": "tour-container",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "tour",
        },
        "ref": null,
        "rendered": "tour",
        "type": "h1",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "cellPadding": "10",
          "cellSpacing": "0",
          "children": Array [
            <thead>
              <tr>
                <th>
                  when
                </th>
                <th>
                  where
                </th>
                <th>
                  wait... where?
                </th>
                <th>
                  how
                </th>
              </tr>
            </thead>,
            <tbody />,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <tr>
                <th>
                  when
                </th>
                <th>
                  where
                </th>
                <th>
                  wait... where?
                </th>
                <th>
                  how
                </th>
              </tr>,
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <th>
                    when
                  </th>,
                  <th>
                    where
                  </th>,
                  <th>
                    wait... where?
                  </th>,
                  <th>
                    how
                  </th>,
                ],
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": "when",
                  "nodeType": "host",
                  "props": Object {
                    "children": "when",
                  },
                  "ref": null,
                  "rendered": "when",
                  "type": "th",
                },
                Object {
                  "instance": null,
                  "key": "where",
                  "nodeType": "host",
                  "props": Object {
                    "children": "where",
                  },
                  "ref": null,
                  "rendered": "where",
                  "type": "th",
                },
                Object {
                  "instance": null,
                  "key": "wait... where?",
                  "nodeType": "host",
                  "props": Object {
                    "children": "wait... where?",
                  },
                  "ref": null,
                  "rendered": "wait... where?",
                  "type": "th",
                },
                Object {
                  "instance": null,
                  "key": "how",
                  "nodeType": "host",
                  "props": Object {
                    "children": "how",
                  },
                  "ref": null,
                  "rendered": "how",
                  "type": "th",
                },
              ],
              "type": "tr",
            },
            "type": "thead",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": undefined,
            },
            "ref": null,
            "rendered": null,
            "type": "tbody",
          },
        ],
        "type": "table",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "No tour dates right now, check back soon!",
        },
        "ref": null,
        "rendered": "No tour dates right now, check back soon!",
        "type": "h3",
      },
    ],
    "type": "section",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <h1>
            tour
          </h1>,
          <table
            cellPadding="10"
            cellSpacing="0"
          >
            <thead>
              <tr>
                <th>
                  when
                </th>
                <th>
                  where
                </th>
                <th>
                  wait... where?
                </th>
                <th>
                  how
                </th>
              </tr>
            </thead>
            <tbody />
          </table>,
          <h3>
            No tour dates right now, check back soon!
          </h3>,
        ],
        "className": "tour-container",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "tour",
          },
          "ref": null,
          "rendered": "tour",
          "type": "h1",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "cellPadding": "10",
            "cellSpacing": "0",
            "children": Array [
              <thead>
                <tr>
                  <th>
                    when
                  </th>
                  <th>
                    where
                  </th>
                  <th>
                    wait... where?
                  </th>
                  <th>
                    how
                  </th>
                </tr>
              </thead>,
              <tbody />,
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <tr>
                  <th>
                    when
                  </th>
                  <th>
                    where
                  </th>
                  <th>
                    wait... where?
                  </th>
                  <th>
                    how
                  </th>
                </tr>,
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": Array [
                    <th>
                      when
                    </th>,
                    <th>
                      where
                    </th>,
                    <th>
                      wait... where?
                    </th>,
                    <th>
                      how
                    </th>,
                  ],
                },
                "ref": null,
                "rendered": Array [
                  Object {
                    "instance": null,
                    "key": "when",
                    "nodeType": "host",
                    "props": Object {
                      "children": "when",
                    },
                    "ref": null,
                    "rendered": "when",
                    "type": "th",
                  },
                  Object {
                    "instance": null,
                    "key": "where",
                    "nodeType": "host",
                    "props": Object {
                      "children": "where",
                    },
                    "ref": null,
                    "rendered": "where",
                    "type": "th",
                  },
                  Object {
                    "instance": null,
                    "key": "wait... where?",
                    "nodeType": "host",
                    "props": Object {
                      "children": "wait... where?",
                    },
                    "ref": null,
                    "rendered": "wait... where?",
                    "type": "th",
                  },
                  Object {
                    "instance": null,
                    "key": "how",
                    "nodeType": "host",
                    "props": Object {
                      "children": "how",
                    },
                    "ref": null,
                    "rendered": "how",
                    "type": "th",
                  },
                ],
                "type": "tr",
              },
              "type": "thead",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": undefined,
              },
              "ref": null,
              "rendered": null,
              "type": "tbody",
            },
          ],
          "type": "table",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "No tour dates right now, check back soon!",
          },
          "ref": null,
          "rendered": "No tour dates right now, check back soon!",
          "type": "h3",
        },
      ],
      "type": "section",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    "disableLifecycleMethods": true,
  },
}
`;
